pkg/ingester/ingester.go

```diff
@@ -556,10 +562,16 @@ func (i *Ingester) LegacyShutdownHandler(w http.ResponseWriter, r *http.Request)
 // ShutdownHandler handles a graceful shutdown of the ingester service and
 // termination of the Loki process.
 func (i *Ingester) ShutdownHandler(w http.ResponseWriter, r *http.Request) {
+	// Don't allow calling the shutdown handler multiple times
+	if i.State() != services.Running {
+		w.WriteHeader(http.StatusNotFound)
+		_, _ = w.Write([]byte("Ingester is stopping or already stopped."))
```

Comment:
Should we `return` here?

Reply:
I'd also argue `404 Not Found` is not the right status code here. Maybe `410 Gone` is better.

Reply:
Indeed :man_facepalming:

Reply:
I was thinking about `410` as well, but I'm not sure if it's better either.
Yes, for this specific instance the endpoint is gone, but after restart, it is available again, making it only temporarily unavailable.
> If you don't know whether this condition is temporary or permanent, a [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) status code should be used instead.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410

Reply:
In our API, `404` is mainly used (AFAIK) when an API endpoint doesn't exist, so I think 404 would be a confusing choice here even if it's the most correct.

Reply:
I thought about it a bit more and I think 503 is more accurate, because the server is not ready to handle the request.

Reply:
That works for me too :+1: